openhands/cli/entry.py

```diff
@@ -0,0 +1,68 @@
+"""Main entry point for OpenHands CLI with subcommand support."""
+
+import asyncio
+import sys
+
+from openhands import __version__
+from openhands.cli.subcommands import (
+    create_subcommand_parser,
+    handle_cli_command,
+    handle_serve_command,
+)
+
+
+def main():
+    """Main entry point with subcommand support and backward compatibility."""
+    # Check if we're being called with subcommands or legacy arguments
+    if len(sys.argv) > 1 and sys.argv[1] in ['serve', 'cli']:
+        # New subcommand interface
+        parser = create_subcommand_parser()
+        args = parser.parse_args()
+
+        if args.version:
+            print(f'OpenHands version: {__version__}')
+            sys.exit(0)
+
+        if args.command == 'serve':
+            handle_serve_command(args)
+        elif args.command == 'cli':
+            # Run CLI mode with asyncio
+            loop = asyncio.new_event_loop()
+            asyncio.set_event_loop(loop)
+            try:
+                loop.run_until_complete(handle_cli_command(args))
+            except KeyboardInterrupt:
+                print('⚠️ Session was interrupted: interrupted\n')
+            except ConnectionRefusedError as e:
+                print(f'Connection refused: {e}')
+                sys.exit(1)
+            except Exception as e:
+                print(f'An error occurred: {e}')
+                sys.exit(1)
+            finally:
+                try:
+                    # Cancel all running tasks
+                    pending = asyncio.all_tasks(loop)
+                    for task in pending:
+                        task.cancel()
+
+                    # Wait for all tasks to complete with a timeout
+                    loop.run_until_complete(
+                        asyncio.gather(*pending, return_exceptions=True)
+                    )
+                    loop.close()
+                except Exception as e:
+                    print(f'Error during cleanup: {e}')
+                    sys.exit(1)
+        else:
+            parser.print_help()
+    else:
+        # Legacy interface - default to CLI mode for backward compatibility
+        # Import the original main function
+        from openhands.cli.main import main as legacy_main
+
+        legacy_main()
```

Comment:
@OpenHands why do we need a legacy main? Can't we just use the same path as "cli" in this case?